### PR TITLE
Improve logging customization [HZ-772]

### PR DESCRIPTION
**Pull request by frant-hartm**
- Provide ability to customize the default pattern via LOGGING_PATTERN
env variable. To support patterns with exception inside the pattern the
default config file now sets alwaysWriteExceptions=false and `%eXe` was
added to the default pattern configuration.

- Provide ability to specify the configuration file via LOGGING_CONFIG
env variable. This takes either relative or absolute path, or file in
config directory. If not set config/log4j2.properties is used.

- Add support for JsonTemplateLayout (log4j-layout-template-json
dependency) and a built in configuration file using JsonLayout.json as
default for the template.

- The json template can be customized via LOGGING_JSON_TEMPLATE, see
https://logging.apache.org/log4j/2.x/manual/json-template-layout.html#event-templates
for out of the box available templates.

- Small cleanup related to default LOGGING_LEVEL - moved the default out
of the startup scripts

Example use:
- Use different pattern:
`LOGGING_PATTERN='{"time":"%date{ISO8601}", "logger": "%logger{36}", "level": "%level",  "msg": "%enc{%m %xEx}{JSON}"}%n' bin/hz start`

- Use different config file:
`LOGGING_CONFIG=log4j2-json.properties bin/hz start`

- Use json config file with another template:
`LOGGING_CONFIG=log4j2-json.properties LOGGING_TEMPLATE="classpath:EcsLayout.json" bin/hz start`

Similarly with docker via `-e ....`

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

**Review by kubamarchwicki — APPROVED** [review]
:tada:

**Review by ldziedziul — COMMENTED** [review]
I believe there's a mix of LOG4J2_CONFIG and LOGGING_CONFIG names in the code and in the description

File: distribution/src/bin-filemode-755/hz [hunk 1]
```diff
@@ -25,6 +25,22 @@ findScriptDir
...
+  LOGGING_CONFIG="file:$HAZELCAST_HOME/config/log4j2.properties"
+else
+  if [ -f "$LOGGING_CONFIG" ]; then
+    # Try if the file exists - either absolute or relative path
+    LOGGING_CONFIG="file:$LOGGING_CONFIG"
+  elif [ -f "$HAZELCAST_HOME/config/$LOGGING_CONFIG" ]; then
+    # Try file in the config directory
+    LOGGING_CONFIG="file:$HAZELCAST_HOME/config/$LOGGING_CONFIG"
+  else
+    echo "Could not find specified LOG4J2_CONFIG file."
```
[the remark this answers is not in the document]
Comment (frant-hartm) [reply distribution/src/bin-filemode-755/hz:39]: Good catch!

**Review by ldziedziul — APPROVED** [review]
Looks Good!

**Pull request merged**